spark-jobs/src/main/scala/filodb/downsampler/chunk/BatchExporter.scala

```diff
@@ -131,9 +133,21 @@ case class BatchExporter(downsamplerSettings: DownsamplerSettings, userStartTime
         getExportData(part, partKeyMap, rule.get)
       }
       .map { exportData =>
-        numRowsExportPrepped.increment()
-        exportDataToRow(exportData, exportTableConfig)
+        try {
+          val row = exportDataToRow(exportData, exportTableConfig)
+          numRowsExportPrepped.increment()
+          Some(row)
+        } catch {
+          case t: Throwable =>
+            if (downsamplerSettings.logAllRowErrors) {
+              DownsamplerContext.dsLogger.error(s"error during exportDataToRow: $exportData", t)
+            }
+            numRowExportPrepErrors.increment()
+            None
+        }
       }
+      .filter(_.isDefined)
```

Comment:
In case of exception, will this throw exception ?

`None.isDefined` ?

Reply:
Nope, `None` is a child type of `Option` -- no NPEs here